### PR TITLE
zephyr: vcp: Add VCP tests

### PR DESCRIPTION
**Pull request by piotrnarajowski**
Adding support for VCP volume controler tests

Merge after[ PR#1002](https://github.com/auto-pts/auto-pts/pull/1002)

### CODE REVIEW

**Comment by piotrnarajowski** [comment]
@KKopyscinski

File: doc/btp_vcp.txt [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+		This command is used to turn the volume up one step and unmute
+		a remote Volume Renderer. During operation, the IUT may send
+		event:
+				VCP State Event
+				VCP Flags Event
+
+	Opcode 0x09 - Set volume
+
+		Controller Index:       <controller id>
+		Command parameters:     Volume (1 octet)
```
Comment (sjanc) [inline doc/btp_vcp.txt:112]: nitpick: address first

Comment (piotrnarajowski) [reply doc/btp_vcp.txt]: Fixed

File: autopts/ptsprojects/zephyr/vcp.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    pts = ptses[0]
+
+    pts.set_pixit("VCP", "TSPX_bd_addr_iut", "DEADBEEFDEAD")
+    pts.set_pixit("VCP", "TSPX_iut_device_name_in_adv_packet_for_random_address", "")
+    pts.set_pixit("VCP", "TSPX_time_guard", "180000")
+    pts.set_pixit("VCP", "TSPX_use_implicit_send", "TRUE")
+    # pts.set_pixit(
+    #     "VCP",
+    #     "TSPX_tester_database_file",
+    #     r"C:\Program Files\Bluetooth SIG\Bluetooth PTS\Data\SIGDatabase\PTS_MICP_db")
```
Comment (mkasenberg) [inline autopts/ptsprojects/zephyr/vcp.py:52]: leftover

File: autopts/pybtp/btp/aics.py [hunk 3]
```diff
@@ -231,96 +231,115 @@ def aics_description_get(bd_addr_type=None, bd_addr=None):
...
+    fmt = '<B6sb'
 
     if len(data) < struct.calcsize(fmt):
         raise BTPError('Invalid data length')
 
-    addr_type, addr = struct.unpack_from(fmt, data)
+    addr_type, addr, att_status = struct.unpack_from(fmt, data)
     addr = binascii.hexlify(addr[::-1]).lower().decode('utf-8')
 
     description = struct.unpack_from(f'<{len(data) - struct.calcsize(fmt) - 1}s', data, offset=struct.calcsize(fmt))
```
Comment (mkasenberg) [inline autopts/pybtp/btp/aics.py:312]:
```suggestion
    description = struct.unpack_from(f'<{len(data) - fmt_size - 1}s', data, offset=fmt_size)
```

File: autopts/pybtp/btp/aics.py [hunk 4]
```diff
@@ -231,96 +231,115 @@ def aics_description_get(bd_addr_type=None, bd_addr=None):
...
+    aics.event_received(defs.AICS_STATUS_EV, (addr_type, addr, status, att_status))
 
 
 def aics_description_ev(aics, data, data_len):
     logging.debug('%s %r', aics_description_ev.__name__, data)
 
-    fmt = '<B6s'
+    fmt = '<B6sb'
 
     if len(data) < struct.calcsize(fmt):
```
Comment (mkasenberg) [inline autopts/pybtp/btp/aics.py:306]:
```suggestion
    fmt_size = struct.calcsize(fmt)
    if len(data) < fmt_size:
```

File: autopts/wid/micp.py [hunk 5]
```diff
@@ -68,10 +68,12 @@ def hdl_wid_5(params: WIDParams):
...
-    ev = stack.aics.wait_aics_state_ev(addr_type, addr, 30)
-    # TODO: Should return False if ev is None
-
-    return True
+    ev = stack.aics.wait_aics_procedure_ev(addr_type, addr, 10)
+    if ev[3] == 2:
+        # opcode == 0x02: set gain
+        return True
+    else:
+        return False
```
Comment (mkasenberg) [inline autopts/wid/micp.py:76]:
```suggestion
    return ev[3] == 2
```
Short the other ones too.

**Review by mkasenberg — CHANGES_REQUESTED** [review]

**Comment by mkasenberg** [comment]
#AutoPTS run zephyr VCP MICP

**Comment by codecoup-tester** [comment]
Scheduled PR #https://github.com/auto-pts/auto-pts/pull/1000#issuecomment-1740767415 after 13:55:59.

**Comment by codecoup-tester** [comment]
AutoPTS Bot results:
<details><summary>Failed tests</summary>MICP    MICP/CL/SPE/BI-01-C             BTP ERROR</details>
<details><summary>Successful tests</summary>MICP    MICP/CL/CGGIT/CHA/BV-01-C       PASS<br>MICP    MICP/CL/CGGIT/CHA/BV-02-C       PASS<br>MICP    MICP/CL/CGGIT/CHA/BV-03-C       PASS<br>MICP    MICP/CL/CGGIT/CHA/BV-04-C       PASS<br>MICP    MICP/CL/CGGIT/CHA/BV-05-C       PASS<br>MICP    MICP/CL/CGGIT/CHA/BV-06-C       PASS<br>MICP    MICP/CL/CGGIT/CHA/BV-07-C       PASS<br>MICP    MICP/CL/CGGIT/SER/BV-01-C       PASS<br>MICP    MICP/CL/CGGIT/SER/BV-02-C       PASS<br>MICP    MICP/CL/CP/BV-01-C              PASS<br>MICP    MICP/CL/CP/BV-02-C              PASS<br>MICP    MICP/CL/CP/BV-03-C              PASS<br>MICP    MICP/CL/CP/BV-04-C              PASS<br>MICP    MICP/CL/CP/BV-05-C              PASS<br>MICP    MICP/CL/SPE/BI-02-C             PASS<br>MICP    MICP/CL/SPE/BI-03-C             PASS<br>MICP    MICP/CL/SPE/BI-04-C             PASS<br>MICP    MICP/CL/SPE/BI-05-C             PASS<br>MICP    MICP/CL/SPE/BI-06-C             PASS<br>MICP    MICP/CL/SPE/BI-07-C             PASS<br>MICP    MICP/CL/SPE/BI-08-C             PASS<br>MICP    MICP/CL/SPE/BI-09-C             PASS<br>MICP    MICP/CL/SPE/BI-10-C             PASS<br>MICP    MICP/CL/SPE/BI-11-C             PASS<br>MICP    MICP/CL/SPE/BI-12-C             PASS<br>VCP     VCP/VC/AICP/BV-01-C             PASS<br>VCP     VCP/VC/AICP/BV-02-C             PASS<br>VCP     VCP/VC/AICP/BV-03-C             PASS<br>VCP     VCP/VC/AICP/BV-04-C             PASS<br>VCP     VCP/VC/AICP/BV-05-C             PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-01-C        PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-02-C        PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-03-C        PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-04-C        PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-05-C        PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-06-C        PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-07-C        PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-08-C        PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-09-C        PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-10-C        PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-11-C        PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-12-C        PASS<br>VCP     VCP/VC/CGGIT/CHA/BV-13-C        PASS<br>VCP     VCP/VC/CGGIT/SER/BV-01-C        PASS<br>VCP     VCP/VC/CGGIT/SER/BV-02-C        PASS<br>VCP     VCP/VC/CGGIT/SER/BV-03-C        PASS<br>VCP     VCP/VC/SPE/BI-01-C              PASS<br>VCP     VCP/VC/SPE/BI-02-C              PASS<br>VCP     VCP/VC/SPE/BI-03-C              PASS<br>VCP     VCP/VC/SPE/BI-04-C              PASS<br>VCP     VCP/VC/SPE/BI-05-C              PASS<br>VCP     VCP/VC/SPE/BI-06-C              PASS<br>VCP     VCP/VC/SPE/BI-07-C              PASS<br>VCP     VCP/VC/SPE/BI-08-C              PASS<br>VCP     VCP/VC/SPE/BI-09-C              PASS<br>VCP     VCP/VC/SPE/BI-10-C              PASS<br>VCP     VCP/VC/SPE/BI-11-C              PASS<br>VCP     VCP/VC/SPE/BI-12-C              PASS<br>VCP     VCP/VC/SPE/BI-13-C              PASS<br>VCP     VCP/VC/SPE/BI-14-C              PASS<br>VCP     VCP/VC/SPE/BI-15-C              PASS<br>VCP     VCP/VC/SPE/BI-16-C              PASS<br>VCP     VCP/VC/SPE/BI-17-C              PASS<br>VCP     VCP/VC/SPE/BI-18-C              PASS<br>VCP     VCP/VC/SPE/BI-19-C              PASS<br>VCP     VCP/VC/SPE/BI-20-C              PASS<br>VCP     VCP/VC/VCCP/BV-01-C             PASS<br>VCP     VCP/VC/VCCP/BV-02-C             PASS<br>VCP     VCP/VC/VCCP/BV-03-C             PASS<br>VCP     VCP/VC/VCCP/BV-04-C             PASS<br>VCP     VCP/VC/VCCP/BV-05-C             PASS<br>VCP     VCP/VC/VCCP/BV-06-C             PASS<br>VCP     VCP/VC/VCCP/BV-07-C             PASS<br>VCP     VCP/VC/VOCP/BV-01-C             PASS</details>

**Review by mkasenberg — APPROVED** [review]

**Pull request merged**